extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -81,19 +81,47 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     return await PassphraseStore.get(this.view.acctEmail, senderKi.fingerprints[0]);
   }
 
-  public lookupPubkeyFromDbOrKeyserverAndUpdateDbIfneeded = async (email: string, name: string | undefined): Promise<Contact | "fail"> => {
+  public lookupPubkeyFromKeyserversThenOptionallyFetchExpiredByFingerprintAndUpsertDb = async (
+    email: string, name: string | undefined
+  ): Promise<Contact | "fail"> => {
+    // note by Tom 2021-08-10: This line below looks weird in the context of recently allowed
+    //    more than one public key per recipient. We are blindly getting the first public key,
+    //    and making decisions based on that. We should be using the whole array instead.
     const [storedContact] = await ContactStore.get(undefined, [email]);
     if (storedContact && storedContact.hasPgp && storedContact.pubkey && !storedContact.revoked) {
-      // Potentially check if pubkey was updated - async. By the time user finishes composing, newer version would have been updated in db.
-      // If sender didn't pull a particular pubkey for a long time and it has since expired, but there actually is a newer version on attester, this may unnecessarily show "bad pubkey",
-      //      -> until next time user tries to pull it. This could be fixed by attempting to fix up the rendered recipient inside the async function below.
+      // checks if pubkey was updated, asynchronously. By the time user finishes composing,
+      //    newer version would have been updated in db.
+      // This implementation is imperfect in that, if sender didn't pull a particular pubkey
+      //    for a long time and the local pubkey has since expired, and there actually is a
+      //    newer version available on external key server, this may unnecessarily show "bad pubkey",
+      //    until next time user tries to enter recipient in the field again, which will at that point
+      //    get the updated key from db. This could be fixed by:
+      //      - either life fixing the UI after this call finishes, or
+      //      - making this call below synchronous and using the result directly
+      //
+      // note by Tom 2021-08-10: I wonder if this functionality could be deprecated, it appears that
```

Comment:
Looks like checkKeyserverForNewerVersionOfKnownPubkeyIfNeeded is using fingerprints, so we should refactor it to accept fingerprints, collected with `getOneWithAllPubkeys`.
`lookupPubkeyFromKeyserversAndUpsertDb` is email-based

Reply:
That's the thing - I think we could and probably should completely drop the fingerprint-based lookup here. 

Let's say I have a `recipient@example.com` with key `ABCD`. Here, we look up by fingerprint to get the same `ABCD` key, and update it. But why do that, when `lookupPubkeyFromKeyserversAndUpsertDb` can look up by `recipient@example.com` and update the key just the same? After recent changes (when we started pulling pubkeys by email on every use), this all of this code that updates by fingerprint seems unneeded. Because what will happen if we just delete this chunk of code? The method `lookupPubkeyFromKeyserversAndUpsertDb` below will get executed, and we'll achieve the same result, just differently.

Reply:
As far as I understand, our storage allows to link an e-mail to arbitrary pubkeys (with missing UserID or having an list of UserIDs etc.), so not all the keys can be found by email on the EKM, unless, of course, an OrgRule forbids to create email->pubkey link manually.

Reply:
This particular functionality was there for customer who doesn't use EKM, and don't have any particular OrgRules affecting this. They have their public keys on our Attester.

Their keys expire, and then get renewed, with the same keys but a newer signature on it. This code would fetch them by fingerprint and get the updated version, but I'm pretty sure if we deleted this `if` and the asynchronous call in it, nothing would change, because the code right below it would update it by email anyway. Does that make sense?

Reply:
Maybe this particular usecase is riddled with too much customer-specific insider info :-) Maybe I'll just try to remove it, and see what happens.